api/prometheus/v1/api.go

```diff
@@ -802,6 +812,29 @@ func (h *httpAPI) TargetsMetadata(ctx context.Context, matchTarget string, metri
 	return res, json.Unmarshal(body, &res)
 }
 
+func (h *httpAPI) MetricsMetadata(ctx context.Context, metric string, limit string) (map[string][]Metadata, error) {
```

Comment:
It might make sense to name the method just `Metadata` so it has the same name as the endpoint.

(nitpick)

Reply:
Agree.  This would also align with the struct name as well.

Reply:
Sounds good to me!

Reply:
Done!